src/v/utils/tests/delta_for_test.cc

```diff
@@ -107,21 +110,57 @@ BOOST_AUTO_TEST_CASE(roundtrip_test_1) {
       100000000000000000ULL,
       1000000000000000000ULL,
     };
-    auto expected = populate_encoder(enc, initial_value, deltas);
+    roundtrip_test<uint64_t>(deltas, details::delta_xor());
+}
 
-    deltafor_decoder<uint64_t> dec(
-      initial_value, enc.get_row_count(), enc.copy());
+BOOST_AUTO_TEST_CASE(roundtrip_test_3) {
+    std::vector<int64_t> deltas = {
+      0LL,
+      10LL,
+      100LL,
+      1000LL,
+      10000LL,
+      100000LL,
+      1000000LL,
+      10000000LL,
+      100000000LL,
+      1000000000LL,
+      10000000000LL,
+      100000000000LL,
+      1000000000000LL,
+      10000000000000LL,
+      100000000000000LL,
+      1000000000000000LL,
+      10000000000000000LL,
+      100000000000000000LL,
+      1000000000000000000LL,
+    };
+    roundtrip_test<int64_t>(deltas, details::delta_delta(0));
+}
 
-    std::vector<uint64_t> actual;
-    std::array<uint64_t, details::FOR_buffer_depth> buf{};
-    int cnt = 0;
-    while (dec.read(buf)) {
-        cnt++;
-        std::copy(buf.begin(), buf.end(), std::back_inserter(actual));
-        buf = {};
-    }
-    BOOST_REQUIRE_EQUAL(cnt, deltas.size());
-    BOOST_REQUIRE(expected == actual);
+BOOST_AUTO_TEST_CASE(roundtrip_test_4) {
+    std::vector<uint64_t> deltas = {
+      0ULL,
+      10ULL,
+      100ULL,
+      1000ULL,
+      10000ULL,
+      100000ULL,
+      1000000ULL,
+      10000000ULL,
+      100000000ULL,
+      1000000000ULL,
+      10000000000ULL,
+      100000000000ULL,
+      1000000000000ULL,
+      10000000000000ULL,
+      100000000000000ULL,
+      1000000000000000ULL,
+      10000000000000000ULL,
+      100000000000000000ULL,
+      1000000000000000000ULL,
+    };
+    roundtrip_test<uint64_t>(deltas, details::delta_delta(0));
```

Comment:
shouldn't we also test with a non-zero step size for the `delta_delta` encoder?

Reply:
fixed it